src/assertions.ts

```diff
@@ -366,6 +369,81 @@ export async function runAssertion({
     };
   }
 
+  if (baseType === 'is-sql') {
+    let parsedSQL;
+    let databaseType: string | undefined = 'MySQL';
+    let whiteTableList: string[] | undefined;
+    let whiteColumnList: string[] | undefined;
+
+    interface sqlParserOption {
+      database: string | undefined;
+      type?: string; // 'type' is optional
+    }
```

Comment:
You might be able to get this type from the `@types/node-sql-parser` package.

Reply:
Thank you for your suggestion. The type here is chosen by me to meet the input requirement of `whiteListCheck` function. When the value of assertion contains a whiteTableList or a whiteColumnList, I will manually put the type as `table` or `column`. The reason why the type is optional is that if the assertion doesn't contain any white list, we don't need the type here anymore. 
I attached two screenshots of the usage of the function `whiteListCheck` for reference. Please let me know if you have any other preferences or suggestions.

![fa2965370baac16f28bf9548f0545a6](https://github.com/promptfoo/promptfoo/assets/113162439/b10359e9-b497-4dec-b1e6-1bdfe25970d8)
![dd1f181ac0e375793933a6d051e6088](https://github.com/promptfoo/promptfoo/assets/113162439/52abc7e7-016e-4dcf-aee8-744365b56dd5)

Reply:
You can import this type from the library itself rather than defining it in multiple places.  For example,
```ts
import { Parser, type Option as SqlParserOption } from 'node-sql-parser';
```